Makefile

```diff
@@ -16,5 +20,11 @@ vendor:
 install-dev:
 	# a tool to vendor non-go files
 	go get github.com/goware/modvendor@latest
+	# a tool for fuzzing
+	go get github.com/dvyukov/go-fuzz/go-fuzz github.com/dvyukov/go-fuzz/go-fuzz-build
 .PHONY: install-dev
 
+fuzz:
+	CGO_ENABLED=0 go-fuzz-build -tags ios github.com/status-im/whisper/whisperv6
```

Comment:
Couple things for context.

The fuzzing tool doesn't like CGO things, so we disabled it here (as it's not necessary anyway for what we're testing.)

The `ios` tag is so that some CPU metrics from `go-ethereum` are disabled in the build (these will fail the build with CGO disabled.)

Reply:
> The ios tag is so that some CPU metrics from go-ethereum are disabled in the build (these will fail the build with CGO disabled.)

Can you elaborate a bit about this?

Reply:
Sure!

On our fork of go-ethereum, we've tagged `metrics/cpu.go` as so not to have it built for the `ios` or `android` tags. This diverges from the upstream implementation, which does not have these tags.

Metrics are captured through the use of a library, [`gosigar`](https://github.com/elastic/gosigar) which has native implementations for things like CPU usage, load average, memory usage, etc, and those features are implemented using native APIs (load average on [darwin](https://github.com/elastic/gosigar/blob/master/sigar_darwin.go#L34) and [linux](https://github.com/elastic/gosigar/blob/master/sigar_linux_common.go#L40) are calculated differently, some of which rely on calling out to C APIs.) As such, they need CGO capabilities to be enabled.

Because we're disabling CGO while building the fuzzing, and this code path still gets compiled, I've added the `ios` tag to make sure it doesn't, as a quick workaround to omitting some code we don't necessarily need.

Does that make more sense?

Reply:
Awesome explanation thanks! I totally forgot about CPU usage being collected by go-ethereum and that obviously needs some C stuff (although syscall maybe could work 🤔 ). Anyway, all clear! I think we can keep it this way and maybe add a comment in Makefile that `ios` tag is to avoid C dependencies. Another option would be a new tag `nometrics` or something to disable C dependencies explicitly but not sure it's worth it.

Reply:
I'd prefer the additional tag so it is more clear, @jakubgs thoughts?